modules/misunderstood/src/hooks/after_incoming_middleware/00_misunderstood_flag.js

```diff
@@ -1,7 +1,27 @@
 const axios = require('axios')
 
 const flag = async () => {
-  if (event.type === 'text' && event.nlu.intent.name === 'none') {
+  const matchesQuickReply = async () => {
+    const { currentFlow } = event.state.context
+    if (currentFlow.startsWith('skills/choice')) {
+      const flow = await bp.ghost.forBot(event.botId).readFileAsObject('flows', currentFlow)
+
+      for (const kwList of Object.values(flow.skillData.keywords)) {
+        for (const kw of kwList) {
+          if (kw.toLowerCase() === event.preview.toLowerCase()) {
+            bp.logger.info(
+              `Misunderstood: event matches Choice Skill quick reply (preview: ${event.preview}, keyword: ${kw})`
+            )
+            return true
+          }
+        }
+      }
+    }
+
+    return false
+  }
+
+  if (event.type === 'text' && event.nlu.intent.name === 'none' && !(await matchesQuickReply())) {
```

Comment:
The matchesQuickReply method should be inside your if, this way we only check if the intent is none / type text.

It should also be wrapped in a try catch so it doesn't impact if there's an error

Reply:
> The matchesQuickReply method should be inside your if, this way we only check if the intent is none / type text.
No, if `(event.type === 'text') === false` or `(event.nlu.intent.name === 'none') === false`, then `matchesQuickReply()` will not be called

> It should also be wrapped in a try catch so it doesn't impact if there's an error
Not sure about this. What errors do you see could happen inside the `matchesQuickReply()` function?

Reply:
You are right. Well you do call the BPFS to fetch a file, so anything can happen.

Reply:
The rest of the hook POSTs to `/mod/misunderstood/events`, which in turn writes to the DB.
Does calling the BPFS really worsen the situation?